src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,309 @@
+    const real_type sin_theta = sqrt((1 - cos_theta) * (1 + cos_theta));
+    const real_type phi = 2 * celeritas::constants::pi * uniform_sample(rng);
+    return {sin_theta * cos(phi), sin_theta * sin(phi), cos_theta};
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculates the form factor based on the form factor model.
+ * TODO: Reference?
+ */
+CELER_FUNCTION real_type WentzelDistribution::calculate_form_factor(
+    real_type formf, real_type cos_t) const
+{
+    switch (data_.form_factor_type)
+    {
+        case NuclearFormFactorType::Flat: {
+            // In units MeV
+            const real_type ccoef = 0.00508;
+            const real_type x = sqrt(2 * inc_mom_sq() * (1 - cos_t)) * ccoef
+                                * 2;
+            return flat_form_factor(x)
+                   * flat_form_factor(
+                       x * 0.6
+                       * fastpow(value_as<Mass>(target_.nuclear_mass()),
+                                 static_cast<real_type>(1) / 3));
+        }
+        break;
+        case NuclearFormFactorType::Exponential:
+            return 1 / ipow<2>(1 + formf * (1 - cos_t));
+        case NuclearFormFactorType::Gaussian:
+            return exp(-2 * formf * (1 - cos_t));
+        default:
+            return 1;
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Flat form factor
+ * TODO: Reference?
+ */
+CELER_FUNCTION real_type WentzelDistribution::flat_form_factor(real_type x) const
+{
+    return 3 * (sin(x) - x * cos(x)) / ipow<3>(x);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Atomic number of the target isotope
+ */
+CELER_FUNCTION int WentzelDistribution::target_Z() const
+{
+    return target_.atomic_number().get();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Helper function to calculate the incident momentum squared.
+ */
+CELER_FUNCTION real_type WentzelDistribution::inc_mom_sq() const
+{
+    return inc_energy_ * (inc_energy_ + 2 * inc_mass_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculates the Moilere screening coefficient.
+ */
+CELER_FUNCTION real_type WentzelDistribution::compute_screening_coefficient() const
+{
+    // TODO: Reference for just proton correction?
+    real_type correction = 1;
+    const real_type sq_cbrt_z = fastpow(static_cast<real_type>(target_Z()),
+                                        static_cast<real_type>(2) / 3);
+    if (target_Z() > 1)
+    {
+        const real_type tau = inc_energy_ / inc_mass_;
+        // TODO: Reference for this factor?
+        const real_type factor = sqrt(tau / (tau + sq_cbrt_z));
+        const real_type inv_beta_sq = 1 + ipow<2>(inc_mass_) / inc_mom_sq();
+
+        correction = min(
+            target_Z() * 1.13,
+            1.13
```

Comment:
clang-format?

Reply:
Yeah that was something clang-format did. I'll try resetting it and see if it formats it nicer.